plugins/Goals/API.php

```diff
@@ -80,10 +80,11 @@ public function getGoal($idSite, $idGoal)
      * Returns all Goals for a given website, or list of websites
      *
      * @param string|array $idSite Array or Comma separated list of website IDs to request the goals for
+     * @param bool $orderByName
      *
      * @return array Array of Goal attributes
      */
-    public function getGoals($idSite)
+    public function getGoals($idSite, bool $orderByName = false): array
```

Comment:
Is there anything you could do to clean up this function/method a bit while you are working in this area?

Reply:
What should be cleaned up in this function? It's actually quite straight forward. As the method is used very often it has a non persistent caching

Reply:
I also had a look and I wasn't sure myself what I'd improve in the method.

Reply:
Well just the method is doing many things like:
- Getting goals, while also setting (in cache)
- Cleaning goals
- Sorting goals

Breaking it into smaller methods would be easier to read, and then if that looks messy having all that logic in the `API` it should be elsewhere.

You don't need to change anything, especially if it's just my preference, just wanted to raise it.

Reply:
Normally I would have put the sorting into the model, but in this special case that would have meant we would need to cache twice - one sorted and one unsorted. So in terms of performance I decided to do the sorting only when needed.